src/invert/laplace/invert_laplace.cxx

```diff
@@ -299,12 +299,6 @@ void Laplacian::tridagCoefs(int jx, int jy, BoutReal kwave,
     localcoords = localmesh->getCoordinates(loc);
   }
 
-  ASSERT1(c1coef == nullptr || c1coef->getLocation() == loc);
-  ASSERT1(c2coef == nullptr || c2coef->getLocation() == loc);
-  ASSERT1( (c1coef == nullptr and c2coef == nullptr)
-           or (c1coef != nullptr and c2coef != nullptr) );
```

Comment:
Is it worth keeping the consistency check for C1 and C2 (maybe as `ASSERT3`)?

Reply:
Maybe? This is a `protected` method, so isn't really callable from user code and we should be able to check library code is calling it correctly. Having it at `ASSERT3` might not hurt though.

Reply:
My thinking is that checking here we have a single point to check, rather than relying on every different implementation of a Laplace solver, etc., to do the right thing. I think they are all correct by design at the moment (Laplace solvers initialise `Field2D`s for all the coefficients, and `Delp2` always passes `nullptr`), but maybe there will be a new implementation in future, and a check here might help catch bugs...

Reply:
This is true :) Unfortunately, it's also called for each mesh point, many, many times, so I was hoping we could avoid any checks in here.

Reply:
Despite `Delp2` using `tridagCoefs` (via `laplace_tridag_coefs`), this usage is deprecated (at least by comments in `invert_laplace.hxx`), so maybe it'd make sense to do the check in `tridagMatrix` - at least that gets the check out of the lowest-level loop.

Reply:
:+1: sounds good

Reply:
I think with some refactoring of `Laplacian` we could eventually remove the need for that check as well.